api/internal/plugins/builtinhelpers/builtins.go

```diff
@@ -100,7 +100,6 @@ var TransformerFactories = map[BuiltinPluginType]func() resmap.TransformerPlugin
 	HashTransformer:                builtins.NewHashTransformerPlugin,
 	ImageTagTransformer:            builtins.NewImageTagTransformerPlugin,
 	LabelTransformer:               builtins.NewLabelTransformerPlugin,
-	LegacyOrderTransformer:         builtins.NewLegacyOrderTransformerPlugin,
```

Comment:
Lines like this one suggest that the old transformer was exposed in the `transformers` field, and I confirmed this experimentally. Here's the setup:

```yaml
# kustomization.yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

resources:
  - resources.yaml # any resources that will show the difference btw fifo and legacy will do

transformers:
  - transformer.yaml
```
```yaml
# transformer.yaml
apiVersion: builtin
kind: LegacyOrderTransformer # the presence of this alone causes legacy order
metadata:
  name: reorder
```
```bash
kustomize build testdir/ --reorder=none # without the transformer, this will be respected. With the transformer, order will be legacy.
```

@natasha41575 do you think we need to preserve this? It was not documented, I don't see any usage of it at all in a cross-Github search, and I don't think it is very useful. So I'm inclined to think it is fine to get rid of as this PR currently does.

Reply:
Happy to get rid of it, I really doubt anyone uses it. If we can get this PR in the next couple weeks, we can also include this as part of the v5 breaking changes.